packages/amplify-prompts/src/progressbars/terminal.ts

```diff
@@ -0,0 +1,94 @@
+const ESC = '\u001b';
+
+const cursorUp = (n : number) : string => {
+  const dy = typeof n === 'number' ? n : 1;
+  return dy > 0 ? `${ESC}[${dy}A` : '';
+};
+
+const clearLine = () : string => `${ESC}[K`;
+
+/**
+ * Amplify Terminal instance (Re writable block)
+ */
+export class AmplifyTerminal {
+    private lastHeight : number;
+    private trailingEmptyLines : number;
+    private readonly stream: NodeJS.WriteStream;
+
+    constructor() {
+      this.lastHeight = 0;
+      this.trailingEmptyLines = 0;
+      this.stream = process.stdout;
+    }
+
+    /**
+     * Checks if the environment is tty
+     */
+    isTTY(): boolean {
+      return this.stream.isTTY;
+    }
+
+    /**
+     * Width of terminal
+     */
+    public get width() : number {
+      return this.stream.columns;
+    }
+
+    /**
+     * Height of last printed block
+     */
+    public getLastHeight() : number {
+      return this.lastHeight;
+    }
+
+    /**
+     * Height of terminal
+     */
+    public get height() : number {
+      return this.stream.rows;
+    }
+
+    /**
+     * Write array of lines into block.
+     */
+    public writeLines(lines : string[]) : void {
+      // Removing /n's since we are accounting for number of lines printed.
+      // This is done to retain cursor positions.
+      const newLines = lines.flatMap(line => line.split('\n'));
+
+      // Go back to beginning of last written block
+      this.stream.write(cursorUp(this.lastHeight));
+
+      newLines.forEach(line => this.stream.write(`${clearLine()}${line}\n`));
```

Comment:
for here and anywhere else that we use `\n` instead of `os.EOL`, we need to make sure that lines are broken properly on windows